frontend/coreproject/src/lib/components/pages/anime_info/AnimeMainDetails.svelte

```diff
@@ -0,0 +1,198 @@
+  import Edit from '$icons/Edit.svelte';
+  import ExternalLink from '$icons/ExternalLink.svelte';
+</script>
+
+<div class="grid grid-cols-12 items-end">
+	<div class="col-span-6 flex items-end gap-10">
+		<img class="w-1/3 rounded-xl" src={anime.cardBackgroundImage} alt={anime.titles.eng} />
+		<div>
+			<h2 class="font-bold">{anime.titles.eng}</h2>
+
+			<div class="anime_titles mt-3 flex items-center gap-2 text-xs font-medium opacity-80">
+				<span>{anime.titles.japanese}</span>
+				<span>-</span>
+				<span>{anime.titles.eng}</span>
+				<span>-</span>
+				<span>{anime.titles.others}</span>
+			</div>
+
+			<div
+				class="anime_basic_details mt-2 flex flex-wrap items-center gap-2 text-xs font-semibold opacity-100 pr-5"
+			>
+				<span>{anime.type}</span>
+				<span>-</span>
+				<span>{anime.episodes} eps</span>
+				<span>-</span>
+				<span>{anime.status}</span>
+				<span>-</span>
+				<span>{anime.premiered}</span>
+				<span>-</span>
+				<span>{anime.studio}</span>
+			</div>
+
+			<div class="mt-5 flex items-center gap-5">
+				<button
+					type="button"
+					class="btn h-16 justify-start rounded-lg bg-primary-500 py-3 pl-4 text-xs font-bold text-white"
+				>
+					<PlayCircle width="30" height="30" color="white" />
+					<div class="flex flex-col items-start">
+						<span class="leading-5">Watch</span>
+						<span class="text-[0.6rem] font-normal leading-none opacity-70">Ep 01</span>
+					</div>
+				</button>
+
+				<button
+					type="button"
+					class="btn h-16 flex-col gap-1 rounded-lg bg-secondary-100 text-xs font-bold text-surface-500"
+				>
+					<Book width="22" height="22" color="bg-surface-500" />
+					Read
+				</button>
+
+				<button
+					type="button"
+					class="btn h-16 flex-col gap-1 rounded-lg bg-secondary-100 text-xs font-bold text-surface-500"
+				>
+					<HeadPhones width="22" height="22" color="bg-surface-500" />
+					Listen
+				</button>
+			</div>
+
+			<div class="user_options mt-5 flex gap-2">
+				<button
+					type="button"
+					class="btn-icon btn-icon-sm rounded bg-warning-400 p-0 text-surface-500"
+				>
+					<Video width="20" />
+				</button>
+
+				<button
+					type="button"
+					class="btn-icon btn-icon-sm rounded bg-warning-400 p-0 text-surface-500"
+				>
+					<Edit3 width="20" />
+				</button>
+
+				<button
+					type="button"
+					class="btn-icon btn-icon-sm rounded bg-warning-400 p-0 text-surface-500"
+				>
+					<Download width="20" />
+				</button>
+
+				<button
+					type="button"
+					class="btn-icon btn-icon-sm rounded bg-warning-400 p-0 text-surface-500"
+				>
+					<Share width="20" />
+				</button>
+			</div>
+		</div>
+	</div>
+
+	<div class="col-span-4 pr-10">
+		<div class="flex items-center gap-4">
+      <h4 class="font-semibold">Synopsis</h4>
+      <SettingsOutline width="15" height="15" class="opacity-75" />
+    </div>
+		<p class="mt-5 text-justify !text-xs opacity-80">
+			{truncate_str(anime.description, 500)}
```

Comment:
whats this function for?

Reply:
It make the string short and add `...`, I think it will be useful on other sections

Reply:
> It make the string short and add ..., I think it will be useful on other sections

Dont reinvent the wheel. Use [voca.js](https://vocajs.com)

Reply:
> Dont reinvent the wheel. Use [`voca.js`](https://vocajs.com)

Do we need to install voca for this

Reply:
> Do we need to install voca for this

Yes